core/vm/common.go

```diff
@@ -87,7 +89,15 @@ func ToWordSize(size uint64) uint64 {
 }
 
 func allZero(b []byte) bool {
-	for _, byte := range b {
+	// 8-byte strides
+	n8 := len(b) - len(b)%8
+	for i := 0; i < n8; i += 8 {
+		if 0 != binary.BigEndian.Uint64(b[i:i+8]) {
```

Comment:
```suggestion
		if 0 != binary.NativeEndian.Uint64(b[i:i+8]) {
```
No need for byteswap...

Reply:
For the curious, the change `BigEndian` to `NativeEndian` here has negligible to no performance hit on x86_64 - I tested